// app/api/aa-lib.js
const axios = require("axios")
const errCodeMap = require("./aa-error-code")

// 本地配置
const localConfig = {
  baseURL: "",
}

// 线下配置
const devConfig = {
  baseURL: "",
}

// 线上配置
const productionConfig = {
  baseURL: "",
}

// 切换线上/线下的配置
// const config = localConfig
const config = devConfig
// const config = productionConfig

let testMap = null
// 如果要用模拟的数据，就打开下面这行的注释
// testMap = require("../test-data")

const axClient = axios.create({
  baseURL: config.baseURL,
  timeout: 10000,
  headers: {
    "Content-Type": "text/plain",
    // "Content-Encoding": "gzip",
  },
  withCredentials: true,
  transformResponse: [
    (resp) => {
      try {
        return JSON.parse(resp)
      } catch (err) {
        throw {
          code: 500,
          msg: "服务器错误",
          data: null,
        }
      }
    },
    (resp) => {
      let code = resp.code
      if (code !== 0) {
        throw {
          code: code,
          msg: errCodeMap.get(code),
          data: resp.data,
        }
      }
      return resp.data
    },
  ],
})

module.exports = {
  ax(path, postData, config) {
    if (testMap) {
      return new Promise((resolve, reject) => {
        window.setTimeout(() => {
          resolve(testMap[path])
        }, 1000)
      })
    }

    return axClient.post(path, postData, config)
      .then((resp) => {
        return resp.data
      })
  },
}

/* 
usage:
  api.GetUser()
    .then((data) => {
      console.log(data)
    })
    .catch((err) => {
      console.log(err)
    })

response:
{
}

request:
{
}
*/
